src/unix/aix.c

```diff
@@ -1077,13 +1078,17 @@ int uv_interface_addresses(uv_interface_address_t** addresses,
   ifr = ifc.ifc_req;
   while ((char*)ifr < (char*)ifc.ifc_req + ifc.ifc_len) {
     p = ifr;
+    inet6 = 0;
     ifr = (struct ifreq*)
       ((char*)ifr + sizeof(ifr->ifr_name) + ADDR_SIZE(ifr->ifr_addr));
 
     if (!(p->ifr_addr.sa_family == AF_INET6 ||
           p->ifr_addr.sa_family == AF_INET))
       continue;
 
+    if (p->ifr_addr.sa_family == AF_INET6)
+      inet6 = 1;
```

Comment:
Simplify to `inet6 = (p->ifr_addr.sa_family == AF_INET6)`?  That also lets you drop the assignment on line 1081.